server/events/post_workflow_hooks_command_runner.go

```diff
@@ -89,13 +96,34 @@ func (w *DefaultPostWorkflowHooksCommandRunner) runHooks(
 	repoDir string,
 ) error {
 
-	for _, hook := range postWorkflowHooks {
-		_, err := w.PostWorkflowHookRunner.Run(ctx, hook.RunCommand, repoDir)
+	for i, hook := range postWorkflowHooks {
+		hookDescription := hook.StepDescription
+		if hookDescription == "" {
+			hookDescription = fmt.Sprintf("Post workflow hook #%d", i)
+		}
 
+		hookId := uuid.New().String()
+		url, err := w.Router.GenerateProjectWorkflowHookURL(hookId)
 		if err != nil {
 			return err
 		}
-	}
 
+		if err := w.CommitStatusUpdater.UpdatePostWorkflowHook(ctx.Pull.BaseRepo, ctx.Pull, models.PendingCommitStatus, hookDescription, "", url); err != nil {
+			ctx.Log.Warn("unable to update post workflow hook status: %s", err)
+		}
+
+		_, runtimeDesc, err := w.PostWorkflowHookRunner.Run(ctx, hook.RunCommand, hookId, repoDir, w.OutputHandler)
+
+		if err != nil {
+			if err := w.CommitStatusUpdater.UpdatePostWorkflowHook(ctx.Pull.BaseRepo, ctx.Pull, models.FailedCommitStatus, hookDescription, runtimeDesc, url); err != nil {
+				ctx.Log.Warn("unable to update post workflow hook status: %s", err)
+			}
+			return err
+		}
+
+		if err := w.CommitStatusUpdater.UpdatePostWorkflowHook(ctx.Pull.BaseRepo, ctx.Pull, models.SuccessCommitStatus, hookDescription, runtimeDesc, url); err != nil {
```

Comment:
Could we move this if block above line 117 and then remove lines 118 to 120?

Reply:
Line 114 marks the VCS check as pending, line 121 VCS marks the check as failed and line 127 marks VCS check as success. I don't think we can remove these lines as it would remove the VCS check update functionality for the workflow hooks.

![image](https://user-images.githubusercontent.com/14815404/207635387-3cf2f1ab-79a3-4471-84ae-ff5f61de0d1d.png)